stix_shifter_modules/gcp_chronicle/stix_transmission/results_connector.py

```diff
@@ -0,0 +1,418 @@
+
+        finally:
+            try:
+                if not return_obj.get('data') or result_count >= self.api_client.result_limit or \
+                        response_dict.get('message') or not response_text.get('nextPageToken'):
+                    if 'code' in response_dict:
+                        if response_dict['code'] not in (1010, 1015):
+                            self.logger.debug("Deleting the search id in results_connector")
+                            self.api_client.delete_search(search_id)
+                    else:
+                        self.logger.debug("Deleting the search id in results_connector")
+                        self.api_client.delete_search(search_id)
+
+            except Exception:
+                self.logger.info("User doesn't have permission to delete the search id")
+        return return_obj
+
+    def invalid_response(self, return_obj, response_dict, status, response_text):
+        """
+        handle error response
+        :return dict
+        """
+
+        response_dict['code'] = status
+        response_dict['message'] = response_text['error'].get('message')
+        ErrorResponder.fill_error(return_obj, response_dict, ['message'],
+                                  connector=self.connector)
+        return return_obj
+
+    @staticmethod
+    def get_results_data(response_text):
+
+        """
+        Format the input results
+        :return list
+        """
+        final_result = []
+
+        for log in response_text:
+            if isinstance(log, list):
+                for element in log:
+                    result_dict = ResultsConnector.get_logs(element)
+                    final_result.append(result_dict)
+            else:
+                result_dict = ResultsConnector.get_logs(log)
+                final_result.append(result_dict)
+        return final_result
+
+    @staticmethod
+    def get_logs(element):
+        """
+        Formats the input with detection and collectionElements
+        :return dict
+        """
+        result_dict = {}
+        event = element['collectionElements'][0]['references'][0]
+        formatted_event = ResultsConnector.format_event_data(event)
+        detection = {"detection": element['detection'][0]}
+        result_dict.update(formatted_event)
+        result_dict.update(detection)
+        return result_dict
+
+    @staticmethod
+    def format_event_data(events):
+        """
+        formats the results data
+        :return dict
+        """
+        for noun in ('principal', 'src', 'target'):
+            if noun in events['event'].keys():
+                events = ResultsConnector.format_registry_result(events, noun)
+                events = ResultsConnector.format_software_result(events, noun)
+                events = ResultsConnector.format_user_account(events, noun)
+        events = ResultsConnector.validate_guardduty_event(events)
```

Comment:
why do you need to validate guardduty events? what happens if you don't?

Reply:
If the guardduty event is not validated, it would provide incorrect source location results in the final stix output.